File: backup.py
import requests
import os
import datetime
from decouple import config
import base64
import json
import re

zendesk_HC_URL = '{}'.format(config('ZENDESK_URL'))
token = '{}'.format(config('TOKEN'))
user = '{}'.format(config('USER'))
credentials = '{user}/token:{token}'.format(user=user, token=token)
language = 'en-US'

date = datetime.date.today()
backup_path = os.path.join(str(date))

if (not os.path.exists(backup_path)):
    os.makedirs(backup_path)

endpoint = zendesk_HC_URL + "/api/v2/help_center/{locale}/articles.json".format(locale=language.lower())

credentials_byte = credentials.encode('ascii')
base_64 = base64.b64encode(credentials_byte)
base64_credentials = base_64.decode('ascii')

while endpoint:
    response = requests.get(endpoint, headers={"Authorization": "Basic {credentials}".format(credentials=base64_credentials)})
    if (response.status_code != 200):
        print('Failed with error {}'.format(response.status_code))
        exit()
    data = response.json()

    for article in data['articles']:
        if (article['body'] is None):
            continue
        title = r"{title}".format(title=article['title'])
        sanitized_title = re.sub(r'[#/?%*{}\\<>\*$!\'":@+`|=]+', ' ', title)
        filename = '{title}.json'.format(title=sanitized_title)

        with open(os.path.join(backup_path, filename), mode='w', encoding='utf-8') as file:
            json.dump(article, file, ensure_ascii=False, indent=4)

    endpoint = data['next_page']

File: backup.js
const axios = require("axios");
const fs = require("fs");
const path = require("path");
require("dotenv").config();

const user = process.env.USER;
const zdURL = process.env.ZENDESK_URL;
const token = process.env.TOKEN;
const locale = "en-US";

function correctdate() {
    const date = new Date(Date.now());
    return `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
}

const date = correctdate();
const auth = Buffer.from(`${user}:token/${token}`).toString("base64");
const endpoint = `${zdURL}/api/v2/help_center/${locale.toLowerCase()}/articles.json`;
let once = 1;

(async function getDataAndWriteToFile(url, filename) {
    function getCorrectPath(title) {
        const p = path.join(__dirname, filename);
        if (!fs.existsSync(p)) {
            fs.mkdirSync(p)
        }
        return `${p}\\${title}.json`;
    }

    while (url) {
        const response = await (await axios.get(url)).data;
        console.log(response.next_page);

        for (let i = 0; i < response.articles.length; i++) {
            const article = response.articles[i];

            if (!article.body) {
                continue;
            }
            const title = article.title.replace(/[#/?%*{}\\<>\*$!\'":@+`|=]+/g, " ");
            fs.writeFileSync(
                getCorrectPath(title),
                JSON.stringify(article),
                function (err) {
                    console.log(err);
                    if (err) return;
                }
            )
        }
        url = response.next_page;
    }
})(endpoint, date);